src/js/wrap.cpp

```diff
@@ -1277,6 +1279,36 @@ namespace ccf::js
     }
   }
 
+  JSValue js_update_runtime_memory_cap(
+    JSContext* ctx, JSValueConst this_val, int argc, JSValueConst* argv)
+  {
+    if (argc != 2)
+    {
+      return JS_ThrowTypeError(
+        ctx, "Passed %d arguments but expected two", argc);
+    }
+
+    int64_t heap_size = 0;
+    if (JS_ToInt64(ctx, &heap_size, argv[0]) < 0)
+    {
+      return JS_EXCEPTION;
+    }
+
+    int64_t stack_size = 0;
+    if (JS_ToInt64(ctx, &stack_size, argv[1]) < 0)
+    {
+      return JS_EXCEPTION;
+    }
+
+    size_t heap = (size_t)heap_size;
+    size_t stack = (size_t)stack_size;
+
+    Runtime::max_heap_size_ = heap;
+    Runtime::max_stack_size_ = stack;
```

Comment:
This doesn't seem thread safe? If the proposal is executed by a governance proposal in thread `t1`, a new JS runtime may be concurrently created by thread `t2`.

Reply:
Done